File: modules/whip-native-spinner/android/src/main/java/io/github/kaminarios/whip/spinner/WhipNativeSpinnerModule.kt
package io.github.kaminarios.whip.spinner

import android.graphics.Color
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.kotlin.types.ColorCompat

class WhipNativeSpinnerModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("WhipNativeSpinner")

    View(WhipAgentSpinnerView::class) {
      Name("WhipAgentSpinnerView")

      Prop("color") { view: WhipAgentSpinnerView, color: Color ->
        view.setSpinnerColor(ColorCompat.toArgb(color))
      }

      Prop("durationMs") { view: WhipAgentSpinnerView, durationMs: Double ->
        view.setRotationDuration(durationMs.toLong())
      }

      Prop("enabled") { view: WhipAgentSpinnerView, enabled: Boolean ->
        view.setAnimationEnabled(enabled)
      }

      Prop("framesPerSecond") { view: WhipAgentSpinnerView, framesPerSecond: Int ->
        view.setFramesPerSecond(framesPerSecond)
      }
    }
  }
}

File: modules/whip-native-spinner/ios/WhipNativeSpinnerModule.swift
import ExpoModulesCore
import UIKit

public final class WhipNativeSpinnerModule: Module {
  public func definition() -> ModuleDefinition {
    Name("WhipNativeSpinner")

    View(WhipAgentSpinnerView.self) {
      Prop("color") { (view, color: UIColor) in
        view.setSpinnerColor(color)
      }

      Prop("durationMs") { (view, durationMs: Double) in
        view.setRotationDuration(durationMs)
      }

      Prop("enabled") { (view, enabled: Bool) in
        view.setAnimationEnabled(enabled)
      }

      Prop("framesPerSecond") { (view, framesPerSecond: Float) in
        view.setFramesPerSecond(framesPerSecond)
      }
    }
  }
}
